ledger-tool/src/main.rs

```diff
@@ -3982,21 +3982,10 @@ fn main() {
                     force_update_to_open,
                     enforce_ulimit_nofile,
                 );
-                let max_height = if let Some(height) = arg_matches.value_of("max_height") {
-                    usize::from_str(height).expect("Maximum height must be a number")
-                } else {
-                    usize::MAX
-                };
-                let start_root = if let Some(height) = arg_matches.value_of("start_root") {
-                    Slot::from_str(height).expect("Starting root must be a number")
-                } else {
-                    0
-                };
-                let num_roots = if let Some(roots) = arg_matches.value_of("num_roots") {
-                    usize::from_str(roots).expect("Number of roots must be a number")
-                } else {
-                    usize::from_str(DEFAULT_ROOT_COUNT).unwrap()
-                };
+
+                let max_height = value_t!(arg_matches, "max_height", usize).unwrap_or(usize::MAX);
+                let start_root = value_t!(arg_matches, "start_root", Slot).unwrap_or(0);
+                let num_roots = value_t_or_exit!(arg_matches, "num_roots", usize);
```

Comment:
This one has a default value so we can do the `_or_exit` variant of the macro:
https://github.com/solana-labs/solana/blob/6db57f81db2c56e2db28e34bf6bc5a206c807db3/ledger-tool/src/main.rs#L2081-L2085